jstz_api/src/text_encoder.rs

```diff
@@ -1,71 +1,249 @@
+        let s = input.unwrap_or_else(|| "".to_string());
+        s.into_bytes().into_boxed_slice()
+    }
+
+    fn encode_into(
+        source: String,
+        destination: &mut Uint8Array,
+    ) -> TextEncoderEncodeIntoResult {
+        let source = source.into_bytes().into_boxed_slice();
+        let mut count = 0u128;
+        for (i, dst_elt) in destination.iter_mut().enumerate() {
+            if let Some(src_elt) = source.get(i) {
+                count += 1;
+                *dst_elt = *src_elt;
+            }
+        }
+        TextEncoderEncodeIntoResult {
+            read: count,
+            written: count,
+        }
+    }
+}
+#[derive(Default, Clone, Trace, Finalize)]
 pub struct TextEncoderApi;
 impl TextEncoderApi {
-    const NAME: &str = "TextEncoder";
-    fn atob(_: &JsValue, args: &[JsValue], _: &mut Context) -> JsResult<JsValue> {
-        let data = args
-            .get_or_undefined(0)
-            .as_string()
-            .ok_or_else(|| JsNativeError::typ().with_message("expected string"))?;
-        let result = TextEncoder::atob(data)?;
-        Ok(result.into())
+    fn encode(_: &JsValue, args: &[JsValue], context: &mut Context) -> JsResult<JsValue> {
+        let arg = match args.get(0) {
+            None => None,
+            Some(arg) => {
+                let arg = arg
+                    .as_string()
+                    .ok_or_else(|| JsNativeError::typ().with_message("expected string"))?
+                    .to_std_string_escaped();
+                Some(arg)
+            }
+        };
+        let result = TextEncoder::encode(arg);
+        let byte_block = result.to_vec();
+        let array_buffer: JsArrayBuffer =
+            JsArrayBuffer::from_byte_block(byte_block, context)?;
+        let result_js = JsUint8Array::from_array_buffer(array_buffer, context)?;
+        Ok(result_js.into())
     }
-    fn btoa(_: &JsValue, args: &[JsValue], _: &mut Context) -> JsResult<JsValue> {
-        let data = args
-            .get_or_undefined(0)
-            .as_string()
-            .ok_or_else(|| JsNativeError::typ().with_message("expected string"))?;
-        let result = TextEncoder::btoa(data)?;
-        Ok(result.into())
+
+    fn encode_into(
+        _: &JsValue,
+        args: &[JsValue],
+        context: &mut Context,
+    ) -> JsResult<JsValue> {
+        let source = args.get_or_undefined(0).try_js_into(context)?;
+        let destination = args.get_or_undefined(1);
+        // 2nd argument is a reference to a Uint8Array to be overwritten
+        let init: JsUint8Array = destination.try_js_into(context)?;
+        // initialize a boxed "destionation buffer" with the transpiled content of the Js object
+        let mut vec: Vec<u8> = Vec::with_capacity(init.length(context)?);
+        for index in 0..init.length(context)? {
+            vec.push(init.at(index as i64, context)?.to_uint8(context)?)
+        }
+        let destination_buffer: &mut Box<[u8]> = &mut Box::from(vec);
```

Comment:
Instead, you should convert the `JsUint8Array` to an `ArrayBuffer`:
https://docs.rs/boa_engine/latest/boa_engine/builtins/array_buffer/struct.ArrayBuffer.html.

Steps:
1. Convert to typed array data using `.as_typed_array()` (https://docs.rs/boa_engine/latest/boa_engine/object/struct.Object.html#method.as_typed_array)
2. Get array buffer view using https://docs.rs/boa_engine/latest/boa_engine/builtins/typed_array/integer_indexed_object/struct.IntegerIndexed.html#method.viewed_array_buffer.
3. Get `ArrayBuffer` from `JsObject` using https://docs.rs/boa_engine/latest/boa_engine/object/struct.Object.html#method.as_array_buffer_mut

You now have access to `array_buffer_data` :)